src/pyhf/modifiers/shapefactor.py

```diff
@@ -118,7 +118,7 @@ def _precompute(self):
         if not self.param_viewer.index_selection:
             return
         tensorlib, _ = get_backend()
-        self.shapefactor_mask = tensorlib.astensor(self._shapefactor_mask)
+        self.shapefactor_mask = tensorlib.astensor(self._shapefactor_mask, dtype="bool")
```

Comment:
same here.. could just wrap the tensorlib.tile call in precompute to do the cast

Reply:
@lukasheinrich I simplified this into one call instead of two, but this still needs to be done as `tile` requires that `self.shapefactor_mask` is a Tensor, so this needs to be done before `tile` gets evaluated.